pkg/maintainer/spv/spv.go

```diff
@@ -49,18 +57,384 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
 		}
 
 		select {
-		case <-time.After(sm.config.RestartBackOffTime):
+		case <-time.After(sm.config.RestartBackoffTime):
 		case <-ctx.Done():
 			return
 		}
 	}
 }
 
 func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
-	logger.Infof("Maintaining SPV proof...")
+	for {
+		if err := sm.proveDepositSweepTransactions(); err != nil {
+			return fmt.Errorf(
+				"error while proving deposit sweep transactions: [%v]",
+				err,
+			)
+		}
+
+		// TODO: Add proving of other type of SPV transactions: redemption
+		// transactions, moving funds transaction, etc.
+
+		select {
+		case <-time.After(sm.config.IdleBackoffTime):
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (sm *spvMaintainer) proveDepositSweepTransactions() error {
+	depositSweepTransactions, err := sm.getUnprovenDepositSweepTransactions()
+	logger.Infof(
+		"Found [%d] unproven deposit sweep transaction(s)",
+		len(depositSweepTransactions),
+	)
+
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get unproven deposit sweep transactions: [%v]",
+			err,
+		)
+	}
+
+	txProofDifficultyFactor, err := sm.spvChain.TxProofDifficultyFactor()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get transaction proof difficulty factor: [%v]",
+			err,
+		)
+	}
+
+	// TODO: Handle a situation in which the block headers in the proof span
+	//       multiple Bitcoin difficulty epochs.
+	requiredConfirmations := uint(txProofDifficultyFactor.Uint64())
+
+	currentEpochDifficulty, previousEpochDifficulty, err :=
+		sm.btcDiffChain.GetCurrentAndPrevEpochDifficulty()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get Bitcoin epoch difficulties: [%v]",
+			err,
+		)
+	}
+
+	for _, transaction := range depositSweepTransactions {
+		// Print the transaction in the same endianness as block explorers do.
+		transactionHashStr := transaction.Hash().Hex(bitcoin.ReversedByteOrder)
+
+		logger.Infof(
+			"Proceeding with deposit sweep proof for transaction [%s]",
+			transactionHashStr,
+		)
+
+		accumulatedConfirmations, err := sm.btcChain.GetTransactionConfirmations(
+			transaction.Hash(),
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get transaction confirmations: [%v]",
+				err,
+			)
+		}
 
-	// TODO: Implement. For now, just wait.
-	<-ctx.Done()
+		if accumulatedConfirmations < requiredConfirmations {
+			// Skip the transaction as it has not accumulated enough
+			// confirmations. It will be proven later.
+			logger.Infof(
+				"Skipped proving deposit sweep transaction [%s] due to "+
+					"transaction not having enough confirmations yet",
+				transactionHashStr,
+			)
```

Comment:
Let's be more specific:

```suggestion
			// Skip the transaction as it has not accumulated enough
			// confirmations. It will be proven later.
			logger.Infof(
				"Skipped proving deposit sweep transaction [%s]; transaction "+
					"has [%v/%v] confirmations",
				transactionHashStr,
				accumulatedConfirmations,
				requiredConfirmations,
			)
```